### PR TITLE
DDF-1588 updated the URLResourceReader to follow redirects by default

### PR DESCRIPTION
**Pull request by jvettraino**
This PR sets the CXF setting to HTTP redirects by default (DDF-1588).  There are unit tests that verify the proper setting of the CFX property to follow redirects.

---
This change is [Reviewable](https://reviewable.io/reviews/codice/ddf/635)

### CODE REVIEW

**Comment by stustison** [comment]
Can you please reformat using the DDF formatter? https://github.com/codice/ddf-support/tree/master/support-checkstyle/src/main/resources after that, LGTM.

**Comment by pklinef** [comment]
Looks like one of the commits is missing the ticket number.  Please amend the commit message to include the ticket number.  Other than that looks good.

**Comment by kcwire** [comment]
LGTM

File: catalog/core/catalog-core-urlresourcereader/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -18,6 +18,11 @@
         <OCD name="URL Resource Reader"
          id="ddf.catalog.resource.impl.URLResourceReader">
                 <AD
+                description="Check the box if you want the Resource Reader to automatically follow server issued redirects (HTTP Response Code 300 series)"
+                name="Follow Server Redirects" id="followRedirects" required="true"
+                type="Boolean" default="true"/>
```
Comment (kcwire) [inline catalog/core/catalog-core-urlresourcereader/src/main/resources/OSGI-INF/metatype/metatype.xml]: To maintain the current behavior can you update this to by `false` / disabled by default?

**Comment by pklinef** [comment]
By default we should only accept redirects from https to https.  We could provider an option to allow insecure redirects that involve http.

**Comment by jvettraino** [comment]
@pklinef @kcwire I posted some thoughts and questions on the Jira ticket about your comments (https://codice.atlassian.net/browse/DDF-1588) - happy to do what makes sense, just not sure what the desired way forward is.  Seems to me like we would want to keep this as simple as possible and then build on it when more use cases come in (I am only aware of the desire to follow all redirects and to not).

**Comment by pklinef** [comment]
@jvettraino I replied to your comment on the Jira ticket.

**Comment by kcwire** [comment]
@jvettraino looks like the testSetRedirect test is now failing.  Fix it and we should be good to merge.

https://codice.atlassian.net/builds/browse/DDF-PRINC-846

**Comment by jvettraino** [comment]
Anything else need to be done for this to be merged?

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java [hunk 2]
```diff
@@ -204,6 +206,30 @@ public void setRootResourceDirectories(Set<String> rootResourceDirectoryPaths) {
...
+        this.followRedirects = redirect;
+    }
+
+    /**
+     * Gets the autoRedirect property
+     * 
+     * @return true if the server issued redirections should be automatically
+     *         followed
+     */
+    public boolean getFollowRedirects() {
```
Comment (troymohl) [inline catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: This should be Boolean to match the setFollowRedirects(Boolean redirect).

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java [hunk 3]
```diff
@@ -95,6 +95,8 @@
 
     private Set<String> rootResourceDirectories = new HashSet<>();
 
+    private boolean followRedirects = false;
```
Comment (troymohl) [inline catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: This should be Boolean to match the setter.

**Comment by kcwire** [comment]
@jvettraino I'll take care of this today - thanks for the contribution!

**Comment by jvettraino** [comment]
@kcwire will you also make the quick change that @troymohl suggested before merging?

**Comment by kcwire** [comment]
@jvettraino - sorry didn't notice that comment. I would have to push a PR to your fork / branch to update it myself - do you mind making the update real quick and rebasing to master.  I rebased it locally and the changes still pass all tests.

**Comment by jvettraino** [comment]
@kcwire I think it should be good to go now, I rebased and squashed into a single commit

**Pull request merged**